{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This python code has been developped by [Taïs Grippa](https://github.com/tgrippa) (Université Libre de Bruxelles). \n",
    "\n",
    "Code developped on Ubuntu 22.04 (Ubuntu Jammy) and GRASS GIS 8.0.2 using the Docker environment [available here](https://github.com/tgrippa/Weaksupervision_Vaihingen)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Define working environment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Import libraries**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0]\n"
     ]
    }
   ],
   "source": [
    "# Import libraries needed for setting parameters of operating system \n",
    "import os\n",
    "import sys\n",
    "import csv\n",
    "import tempfile\n",
    "import glob\n",
    "import math\n",
    "import pickle\n",
    "import time \n",
    "print(sys.version)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "## Import multiprocessing and functools libraries\n",
    "import multiprocessing\n",
    "from multiprocessing import Pool\n",
    "from functools import partial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Add folder with SCR provided belong to this notebook**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Add local module to the path\n",
    "src = os.path.abspath(os.path.join(os.environ['HOME'],'github','SRC'))\n",
    "if src not in sys.path:\n",
    "    sys.path.append(src)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Setup environment variables for TAIS DESKTOP (Linux Mint + GRASS Dev)**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Please edit the file in `../SRC/config.py`, containing the configuration parameters, according to your own computer setup. The following cell is used to run this file.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "exec(open(os.path.join(os.environ['HOME'],'github','SRC', 'config.py')).read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'GISBASE': '/usr/lib/grass78', 'PYTHONLIB': '/usr/bin/python3', 'gisdb': '/home/tais/GRASSDATA', 'location': 'flair-one', 'permanent_mapset': 'PERMANENT', 'locationepsg': '2154', 'outputfolder': '/home/tais/result', 'inputdir': '/home/tais/data'}\n"
     ]
    }
   ],
   "source": [
    "print(config_parameters)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'legend': '/home/tais/github/Legend.txt'}\n"
     ]
    }
   ],
   "source": [
    "print(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Import functions that setup the environmental variables\n",
    "import environ_variables as envi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PATH\t= /.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/grass78/bin:/usr/lib/grass78/script:/usr/lib/grass78/lib \t\n",
      "HOSTNAME\t= d35fee4c10cd \t\n",
      "DISPLAY\t= unix \t\n",
      "LANG\t= C.UTF-8 \t\n",
      "LC_ALL\t= C.UTF-8 \t\n",
      "JUPYTER_ENABLE_LAB\t= yes \t\n",
      "TINI_VERSION\t= v0.6.0 \t\n",
      "HOME\t= /home/tais \t\n",
      "GIT_PYTHON_REFRESH\t= quiet \t\n",
      "PYDEVD_USE_FRAME_EVAL\t= NO \t\n",
      "JPY_SESSION_NAME\t= /home/tais/github/GRASSGIS/notebook/B_OBIA_RuleBased.ipynb \t\n",
      "JPY_PARENT_PID\t= 7 \t\n",
      "TERM\t= xterm-color \t\n",
      "CLICOLOR\t= 1 \t\n",
      "PAGER\t= cat \t\n",
      "GIT_PAGER\t= cat \t\n",
      "MPLBACKEND\t= module://matplotlib_inline.backend_inline \t\n",
      "PYTHONPATH\t= :/usr/lib/grass78/etc/python:/usr/lib/grass78/etc/python/grass:/usr/lib/grass78/etc/python/grass/script \t\n",
      "LD_LIBRARY_PATH\t= :/usr/lib/grass78/lib \t\n",
      "GISBASE\t= /usr/lib/grass78 \t\n",
      "PYTHONLIB\t= /usr/bin/python3 \t\n",
      "GIS_LOCK\t= $$ \t\n",
      "GISRC\t= /home/tais/.grass7/rc \t\n"
     ]
    }
   ],
   "source": [
    "# Set environmental variables\n",
    "envi.setup_environmental_variables() \n",
    "# Display current environment variables of your computer\n",
    "envi.print_environmental_variables()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**GRASS GIS Python libraries**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Import libraries needed to launch GRASS GIS in the jupyter notebook\n",
    "import grass.script.setup as gsetup\n",
    "# Import libraries needed to call GRASS using Python\n",
    "import grass.script as gscript"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Other functions**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from grass.script import vector\n",
    "# Import function that check existance and create GRASS GIS database folder if needed\n",
    "from grass_database import check_gisdb, check_location, check_mapset, working_mapset\n",
    "# Import functions for processing time information\n",
    "from processing_time import start_processing, print_processing_time\n",
    "# Import function that generate a random name in the GRASS GIS environement\n",
    "from random_layer_name import random_layer_name\n",
    "# Import function that check and create folder\n",
    "from mkdir import check_create_dir\n",
    "# Import function that check if GRASS GIS add-on is installed and install it if needed\n",
    "from gextension import check_install_addon\n",
    "# Import function for sorting string number naturally\n",
    "from sorting_natural import natural_keys"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def launch_mapset(mapset):\n",
    "    #Declare empty list that will contain the messages to return\n",
    "    return_message = []\n",
    "    # Init\n",
    "    gsetup.init(config_parameters['GISBASE'], config_parameters['gisdb'], config_parameters['location'], mapset)\n",
    "    # Check if the location exists and create it if not, with the CRS defined by the epsg code \n",
    "    return_message.append(check_location(config_parameters[\"gisdb\"],config_parameters['location'],config_parameters[\"locationepsg\"]))\n",
    "    # Check if mapset exists\n",
    "    return_message.append(check_mapset(config_parameters[\"gisdb\"],config_parameters['location'],mapset))\n",
    "    # Change the current working GRASS GIS session mapset\n",
    "    return_message.append(working_mapset(config_parameters[\"gisdb\"],config_parameters['location'],mapset))\n",
    "    # Return\n",
    "    return return_message"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def segmentation(mapset):\n",
    "    # Image ID\n",
    "    image_id = int(mapset.split('_')[-1])\n",
    "    # Define computational region\n",
    "    gscript.run_command('g.region', overwrite=True, raster='red', save=\"region\")\n",
    "    # Define group of layers\n",
    "    gscript.run_command('g.remove', quiet=True, flags='f', type='group', name='top')\n",
    "    gscript.run_command('i.group', quiet=True, group='top', input=\"nir,red,green,blue\")\n",
    "    # gscript.run_command('i.group', group='top', input=\"nir,red,green,blue,ndsm\") # TODO: Test of nDSM in the segmentation  \n",
    "    # Unsupervised Segmentation parameter optimization \n",
    "    gscript.run_command('i.segment.uspo', quiet=True, overwrite=True, group='top', segment_map='best_segment', regions='region', segmentation_method='region_growing', \n",
    "                        threshold_start='0.005', threshold_stop='0.05', threshold_step='0.002', minsizes='50', memory='3000', processes='25')\n",
    "    # Trick the segment ID to be sure the ID is unique through all the images\n",
    "    #formula = \"segmentation = (%s*1000000) + best_segment_region_rank1\"%image_id\n",
    "    formula = \"segmentation = best_segment_region_rank1\"\n",
    "    gscript.mapcalc(formula, quiet=True, overwrite=True)\n",
    "    # Exportation\n",
    "    gscript.run_command('r.out.gdal', quiet=True, overwrite=True, flags='m', input='segmentation', \n",
    "                        output=os.path.join(config_parameters['outputfolder'],'segment_rast','segment_%s.tiff'%mapset), format='GTiff')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def compute_stats(mapset):\n",
    "    # Define computational region\n",
    "    gscript.run_command('g.region', overwrite=True, raster='segmentation')\n",
    "    # Define list of raster on which to compute statistics\n",
    "    rast_layers=[]\n",
    "    rast_layers.append('red')\n",
    "    rast_layers.append('green')\n",
    "    rast_layers.append('blue')\n",
    "    rast_layers.append('nir')\n",
    "    rast_layers.append('ndvi')\n",
    "    rast_layers.append('ndsm')\n",
    "    rast_layers.append('text_green_DE')\n",
    "    rast_layers.append('text_green_Entr')\n",
    "    rast_layers.append('text_red_ASM')\n",
    "    rast_layers.append('text_red_IDM')\n",
    "    rast_layers.append('text_nir_DE')\n",
    "\n",
    "    try:\n",
    "        # Compute segment statistics\n",
    "        gscript.run_command('i.segment.stats', quiet=True, overwrite=True, map='segmentation',\n",
    "                            rasters=','.join(rast_layers), raster_statistics='stddev,coeff_var,sum,median,perc_90', \n",
    "                            area_measures='area,perimeter,compact_circle,compact_square,fd',\n",
    "                            csvfile=os.path.join(config_parameters['outputfolder'],'stats', 'stats_%s.csv'%mapset),\n",
    "                            vectormap='segment',\n",
    "                            processes='20')   \n",
    "    except:\n",
    "        #print(\"execution of i.segment.stats failed using Python run_command function. Now try with bash script\")\n",
    "        # Create a temporary file\n",
    "        with tempfile.NamedTemporaryFile(suffix=\".sh\", delete=False) as temp_file:\n",
    "            # Write the bash script content to the temporary file\n",
    "            temp_file.write(b'#!/bin/bash\\n')\n",
    "            content = 'grass /home/tais/GRASSDATA/flair-one/%s --exec i.segment.stats --o --q map=segmentation rasters=%s \\\n",
    "            raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd \\\n",
    "            csvfile=%s vectormap=segment processes=20\\n'%(mapset,','.join(rast_layers),os.path.join(config_parameters['outputfolder'],'stats', 'stats_%s.csv'%mapset)) \n",
    "            temp_file.write(content.encode())\n",
    "        \n",
    "            # Close the temporary file\n",
    "            temp_file.close()\n",
    "            # Get the path of the temporary file\n",
    "            temp_file_path = temp_file.name\n",
    "        \n",
    "            try:\n",
    "                # Add execution permissions to the temporary file\n",
    "                os.chmod(temp_file_path, 0o755)\n",
    "                # Execute the temporary file\n",
    "                stdout = subprocess.run(['bash', temp_file_path])\n",
    "                #print(stdout)        \n",
    "            \n",
    "            except:\n",
    "                print(\"i.segment.stats using bash script failed too. Please check\")\n",
    "        \n",
    "            finally:\n",
    "                # Clean up the temporary file\n",
    "                os.remove(temp_file_path)\n",
    "    # Export vector layer as GPKG\n",
    "    gscript.run_command('v.out.ogr', quiet=True, overwrite=True, input='segment', \n",
    "                        output=os.path.join(config_parameters['outputfolder'],'segment_vect', 'segment_%s.gpkg'%mapset), format='GPKG')"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {
    "tags": []
   },
   "source": [
    "import subprocess\n",
    "\n",
    "def cleaning_tmpfiles(mapset):\n",
    "    stdout = subprocess.run(['grass','--text',os.path.join(config_parameters['gisdb'],config_parameters['location'],mapset)], \n",
    "                            shell=True, capture_output=True, text=True)\n",
    "    stdout = subprocess.run(['rm', '-r', os.path.join(config_parameters['gisdb'],config_parameters['location'],mapset,'vector')])\n",
    "    stdout = subprocess.run(['mkdir', os.path.join(config_parameters['gisdb'],config_parameters['location'],mapset,'vector')])\n",
    "    return stdout"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import subprocess\n",
    "\n",
    "def worker(mapset):\n",
    "    # Launch a new mapset for this image\n",
    "    launch_mapset(mapset)\n",
    "    # Segmentation\n",
    "    segmentation(mapset)\n",
    "    # Compute statistics\n",
    "    compute_stats(mapset)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Create new directories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The folder '/home/tais/result' already exists\n",
      "The folder '/home/tais/result/stats' already exists\n",
      "The folder '/home/tais/result/segment_rast' already exists\n",
      "The folder '/home/tais/result/segment_vect' already exists\n"
     ]
    }
   ],
   "source": [
    "# Check and create folder if needed\n",
    "check_create_dir(config_parameters['outputfolder'])\n",
    "check_create_dir(os.path.join(config_parameters['outputfolder'],'stats'))\n",
    "check_create_dir(os.path.join(config_parameters['outputfolder'],'segment_rast'))\n",
    "check_create_dir(os.path.join(config_parameters['outputfolder'],'segment_vect'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Segmentation (region growing)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import fnmatch\n",
    "\n",
    "def find_files(path, pattern):\n",
    "    for root, dirs, files in os.walk(path):\n",
    "        for file in fnmatch.filter(files, pattern):\n",
    "            yield os.path.join(root, file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 2950 train images\n"
     ]
    }
   ],
   "source": [
    "# Get a list of path to train images\n",
    "train_folder = os.path.join(config_parameters['inputdir'],'train')\n",
    "train_tupple = [(os.path.split(x)[-1].split('.tif')[0],x) for x in find_files(train_folder, \"IMG_*.tif\")]\n",
    "print(f'There are {len(train_tupple)} train images')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "list_mapsets = os.listdir(os.path.join(config_parameters['gisdb'],config_parameters['location']))\n",
    "list_mapsets.remove('PERMANENT')"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "list_mapsets = ['IMG_027772', 'IMG_027782', 'IMG_027769', 'IMG_027778', 'IMG_027766', 'IMG_027776', 'IMG_027787', 'IMG_027784', 'IMG_027783', 'IMG_027779']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 2950 mapsets\n"
     ]
    }
   ],
   "source": [
    "print(f'There are {len(list_mapsets)} mapsets')"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {
    "tags": []
   },
   "source": [
    "start_parallel = start_processing()\n",
    "ncores = 10\n",
    "p = Pool(ncores)\n",
    "output = p.map(cleaning_tmpfiles, list_mapsets[:]) # Cleaning temp files in the mapsets\n",
    "p.close()\n",
    "p.join()\n",
    "# Print\n",
    "print_processing_time(start_parallel, \"Computation (on %s cores) achieved in \"%(ncores))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_027654/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_027654/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_031426/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_031426/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_031426/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_031936/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_031936/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_029097/.tmp/d35fee4c10cd/741730.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029097.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031374/.tmp/d35fee4c10cd/744319.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031514/.tmp/d35fee4c10cd/745034.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031374.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031514.csv vectormap=segment processes=20> ...\n",
      "WARNING: Vector map <segment> already exists and will be overwritten\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031426/.tmp/d35fee4c10cd/747420.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031426.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031727/.tmp/d35fee4c10cd/747650.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028648/.tmp/d35fee4c10cd/748348.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031936/.tmp/d35fee4c10cd/749128.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_027654/.tmp/d35fee4c10cd/747256.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028077/.tmp/d35fee4c10cd/747648.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031727.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028648.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031936.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028077.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027654.csv vectormap=segment processes=20> ...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029097.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031374.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031514.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_032044/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031426.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028077.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028648.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031727.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027654.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031936.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "ERROR: ERROR: ERROR: UnableUnable  toto  makemakeUnable  mapsetmapset   toelementelement   make.tmp/d35fee4c10cd.tmp/d35fee4c10cd mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028252/.tmp/d35fee4c10cd): \n",
      "       File(/home/tais/GRASSDATA/flair-one/IMG_028252/.tmp/d35fee4c10cd):\n",
      "       \n",
      "        (/home/tais/GRASSDATA/flair-one/IMG_028252/.tmp/d35fee4c10cd):existsFile \n",
      "\n",
      "       Fileexists\n",
      "       \n",
      "exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028252/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: UnableERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028252/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      " to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028252/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_031056/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028473/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028473/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028473/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028473/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028473/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_032044/.tmp/d35fee4c10cd/833380.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032044.csv vectormap=segment processes=20> ...\n",
      "WARNING: Vector map <segment> already exists and will be overwritten\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_029103/.tmp/d35fee4c10cd/916318.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029103.csv vectormap=segment processes=20> ...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032044.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028252/.tmp/d35fee4c10cd/951545.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031079/.tmp/d35fee4c10cd/953929.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028252.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031079.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_029197/.tmp/d35fee4c10cd/956377.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_029190/.tmp/d35fee4c10cd/956442.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029197.csv vectormap=segment processes=20> ...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029190.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028473/.tmp/d35fee4c10cd/957877.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031993/.tmp/d35fee4c10cd/957203.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031056/.tmp/d35fee4c10cd/957921.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028473.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031993.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031056.csv vectormap=segment processes=20> ...\n",
      "WARNING: No data base element files found\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029103.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028252.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031079.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028473.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "WARNING: No data base element files found\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029190.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029197.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031993.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031056.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_027612/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: Vector map <segment> already exists and will be overwritten\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028193/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031589/.tmp/d35fee4c10cd/1033345.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031589.csv vectormap=segment processes=20> ...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031589.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031597/.tmp/d35fee4c10cd/1130225.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031597.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_027610/.tmp/d35fee4c10cd/1160992.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027610.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_032284/.tmp/d35fee4c10cd/1171100.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_027612/.tmp/d35fee4c10cd/1169250.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032284.csv vectormap=segment processes=20> ...\n",
      "WARNING: No data base element files found\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027612.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_032213/.tmp/d35fee4c10cd/1175529.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028563/.tmp/d35fee4c10cd/1178056.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032213.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028563.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028193/.tmp/d35fee4c10cd/1183723.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028193.csv vectormap=segment processes=20> ...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031597.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "ERROR: Unable to start Cleaning up temporary files...\n",
      "driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_032403/.tmp/d35fee4c10cd/1190895.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032403.csv vectormap=segment processes=20> ...\n",
      "WARNING: Vector map <segment> already exists and will be overwritten\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032284.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027610.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027612.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032213.csv vectormap=segment processes=20> finished.\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028563.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028193.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032403.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028538/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028538/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031049/.tmp/d35fee4c10cd/1234007.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031049.csv vectormap=segment processes=20> ...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031049.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_029254/.tmp/d35fee4c10cd/1309122.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029254.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_032013/.tmp/d35fee4c10cd/1318686.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032013.csv vectormap=segment processes=20> ...\n",
      "WARNING: No data base element files found\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029254.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031087/.tmp/d35fee4c10cd/1397779.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031087.csv vectormap=segment processes=20> ...\n",
      "WARNING: Vector map <segment> already exists and will be overwritten\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032013.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_032630/.tmp/d35fee4c10cd/1400518.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028410/.tmp/d35fee4c10cd/1400838.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028410.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032630.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028538/.tmp/d35fee4c10cd/1401637.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028888/.tmp/d35fee4c10cd/1401847.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028943/.tmp/d35fee4c10cd/1402184.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028538.csv vectormap=segment processes=20> ...\n",
      "WARNING: No data base element files found\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028888.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028943.csv vectormap=segment processes=20> ...\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_027712/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031087.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032630.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028888.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028538.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028410.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028943.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031381/.tmp/d35fee4c10cd/1452632.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_031031/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_031031/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_031031/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_031031/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031381.csv vectormap=segment processes=20> ...\n",
      "WARNING: No data base element files found\n",
      "ERROR: ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_031173/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_031173/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_032334/.tmp/d35fee4c10cd/1479606.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032334.csv vectormap=segment processes=20> ...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031381.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_027712/.tmp/d35fee4c10cd/1505867.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027712.csv vectormap=segment processes=20> ...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032334.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "WARNING: No data base element files found\n",
      "WARNING: Vector map <segment> already exists and will be overwritten\n",
      "WARNING: No data base element files found\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027712.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_027646/.tmp/d35fee4c10cd/1610590.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_032161/.tmp/d35fee4c10cd/1610308.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027646.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032161.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031031/.tmp/d35fee4c10cd/1612001.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028910/.tmp/d35fee4c10cd/1612708.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028910.csv vectormap=segment processes=20> ...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031031.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031173/.tmp/d35fee4c10cd/1614716.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031173.csv vectormap=segment processes=20> ...\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_032691/.tmp/d35fee4c10cd/1614618.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032691.csv vectormap=segment processes=20> ...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027646.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032161.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028910.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031173.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031031.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032691.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_029243/.tmp/d35fee4c10cd/1664990.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029243.csv vectormap=segment processes=20> ...\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_027713/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: ERROR: Unable to makeUnable  mapsetto  elementmake  .tmp/d35fee4c10cdmapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028704/.tmp/d35fee4c10cd): File\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028704/.tmp/d35fee4c10cd):\n",
      "        existsFile\n",
      "\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028704/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028704/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_032254/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make ERROR: ERROR: Unable to makemapset  mapsetelement  element.tmp/d35fee4c10cd .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_032254/.tmp/d35fee4c10cd): File\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_032254/.tmp/d35fee4c10cd):\n",
      "       exists\n",
      " File\n",
      "       exists\n",
      "Unable toERROR:  makeERROR:  mapsetUnable element  to.tmp/d35fee4c10cd make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_032254/.tmp/d35fee4c10cd): File\n",
      "       \n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_032254/.tmp/d35fee4c10cd):exists\n",
      " File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_032254/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_032254/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable toERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_032254/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      " make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_032254/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_032254/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028502/.tmp/d35fee4c10cd/1676480.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028502.csv vectormap=segment processes=20> ...\n",
      "WARNING: No data base element files found\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029243.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "WARNING: No data base element files found\n",
      "WARNING: Vector map <segment> already exists and will be overwritten\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028502.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028531/.tmp/d35fee4c10cd/1715137.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028531.csv vectormap=segment processes=20> ...\n",
      "WARNING: No data base element files found\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028531.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_027713/.tmp/d35fee4c10cd/1802114.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027713.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_032254/.tmp/d35fee4c10cd/1809600.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_028704/.tmp/d35fee4c10cd/1811987.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032254.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028704.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_032198/.tmp/d35fee4c10cd/1819218.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032198.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_027595/.tmp/d35fee4c10cd/1835929.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031518/.tmp/d35fee4c10cd/1836915.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027595.csv vectormap=segment processes=20> ...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031518.csv vectormap=segment processes=20> ...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_028704.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027713.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032254.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_029105/.tmp/d35fee4c10cd/1852809.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_032198.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029105.csv vectormap=segment processes=20> ...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031518.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "WARNING: No data base element files found\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_027595.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_027686/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028371/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_029145/.tmp/d35fee4c10cd/1896329.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "WARNING: No data base element files found\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029145.csv vectormap=segment processes=20> ...\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_027865/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_027865/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028705/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028705/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028705/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028705/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable toERROR:  makeUnable  mapsetto  elementmake  .tmp/d35fee4c10cdmapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028705/.tmp/d35fee4c10cd): File\n",
      "       \n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028705/.tmp/d35fee4c10cd):exists \n",
      "File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028705/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029105.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "WARNING: Vector map <segment> already exists and will be overwritten\n",
      "WARNING: No data base element files found\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028317/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to make mapset element .tmp/d35fee4c10cd\n",
      "       (/home/tais/GRASSDATA/flair-one/IMG_028317/.tmp/d35fee4c10cd): File\n",
      "       exists\n",
      "ERROR: Unable to start driver <(null)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 484, in <module>\n",
      "    main()\n",
      "  File \"/home/tais/.grass7/addons/scripts/i.segment.stats\", line 465, in main\n",
      "    gscript.run_command(\"db.execute\", input=insert_sql, quiet=True)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 441, in run_command\n",
      "    return handle_errors(returncode, returncode, args, kwargs)\n",
      "  File \"/usr/lib/grass78/etc/python/grass/script/core.py\", line 342, in handle_errors\n",
      "    raise CalledModuleError(module=None, code=code,\n",
      "grass.exceptions.CalledModuleError: Module run None db.execute --q input=/home/tais/GRASSDATA/flair-one/IMG_031534/.tmp/d35fee4c10cd/1913553.1 ended with error\n",
      "Process ended with non-zero return code 1. See errors in the (error) output.\n",
      "Starting GRASS GIS...\n",
      "Cleaning up temporary files...\n",
      "Executing <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_031534.csv vectormap=segment processes=20> ...\n",
      "Execution of <i.segment.stats --o --q map=segmentation rasters=red,green,blue,nir,ndvi,ndsm,text_green_DE,text_green_Entr,text_red_ASM,text_red_IDM,text_nir_DE raster_statistics=stddev,coeff_var,sum,median,perc_90 area_measures=area,perimeter,compact_circle,compact_square,fd csvfile=/home/tais/result/stats/stats_IMG_029145.csv vectormap=segment processes=20> finished.\n",
      "Cleaning up default sqlite database ...\n",
      "Cleaning up temporary files...\n",
      "WARNING: No data base element files found\n"
     ]
    }
   ],
   "source": [
    "# Launch processes in parallel\n",
    "start_parallel = start_processing()\n",
    "ncores = 10\n",
    "p = Pool(ncores)\n",
    "output = p.map(worker, list_mapsets[:])  # Launch the processes for as many items in the list (if function with a return, the returned results are ordered thanks to 'map' function)\n",
    "p.close()\n",
    "p.join()\n",
    "# Print\n",
    "print_processing_time(start_parallel, \"Computation (on %s cores) achieved in \"%(ncores))"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "source": [
    "worker(\"IMG_027779\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
